packages/lib-react-components/src/AnimatedNumber/AnimatedNumber.js

```diff
@@ -4,80 +4,80 @@ import { interpolate } from '@visx/vendor/d3-interpolate'
-        return t => {
-          const interpolatedValue = interpolator(t)
-          if (interpolatedValue === value) setAnimated(true) // animation complete!
-          const niceValue = formatValue(interpolatedValue)
-          return niceValue
-        }
-      })
+/**
+ * Use d3 to animate `element` from `initialValue` to `value` over `duration` in milliseconds.
+ * @param {HTMLElement} element 
+ * @param {number} initialValue 
+ * @param {number} value
+ * @param {number} duration 
+ * @returns a Promise that resolves once the animation is complete.
+ */
+function animateValue(element, initialValue, value, duration) {
+  if (value === initialValue) {
+    return
   }
-
-  function lessAnimation() {
-    select(numRef.current)
+  const interpolator = interpolate(initialValue, value)
+  return select(element)
     .data([value])
     .transition()
-    .duration(0)
-    .textTween(() => {
-      return () => {
-        setAnimated(true) // animation complete!
-        return formatValue(value)
-      }
-    })
-  }
+    .duration(duration)
+    .textTween(() => t => formatValue(interpolator(t)))
+    .end()
+}
 
-  function formatValue(num) {
-    return format(',d')(num)
-  }
+function AnimatedNumber({ duration = 1000, value = 0 }) {
+  const numRef = useRef(null)
+  const [initialValue, setInitialValue] = useState(0)
+  const [animated, setAnimated] = useState(false)
+  // render the value directly after the animation is complete.
+  const displayedValue = animated ? value : initialValue
 
   useEffect(() => {
-    // If we already animated the number once, don't observe intersection
-    // This could happen if the value prop updates, but page did not refresh
-    if (animated) return
-
+    /*
+     * We're animating the DOM node content outside of the React render cycle
+     * so use node.textContent to check if we've already rendered the value.
+     */
     const numElement = numRef.current
+    if (animated) return // only run the intersection observer once.
+    if (formatValue(value) === numElement.textContent) return // nothing to animate yet.
 
-    const intersectionObserver = new window.IntersectionObserver(entries => {
+    const intersectionObserver = new window.IntersectionObserver(async entries => {
       // If intersectionRatio is 0, the target is out of view and we do not need to do anything.
       if (entries[0].intersectionRatio <= 0) return
 
-      // Once target element is in viewport, animate it then unobserve
-      if (!prefersReducedMotion && !animated) {
-        animateValue()
-        intersectionObserver.unobserve(numElement)
-      } else {
-        lessAnimation()
-        intersectionObserver.unobserve(numElement)
-      }
+      /*
+       * Once target element is in viewport, animate it then unobserve.
+       * Only run the full animation if prefers reduced motion is false
+       * and the displayed value is 0.
+       */
+      const isZero = numElement.textContent === formatValue(initialValue)
```

Comment:
The `UpdateTheValue` tests uncovered a race condition where d3 had inserted '10' into `numElement` but `animated` was false, so the animation ran once from 0 to 10, then again from 0 to 11. Testing the value that's been inserted by d3 guards against this and runs the test as expected.

Reply:
You can reproduce that bug in the Storybook, if you click the Update button while the animation is running.

Reply:
I've changed the behaviour of the component so that an animation will always run to completion, even if `value` changes while the animation is running.